Classes/Infrastructure/Healthcheck/BasicRequirementsHealthcheck.php

```diff
@@ -0,0 +1,287 @@
+<?php
+namespace Neos\Setup\Infrastructure\Healthcheck;
+
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Core\Bootstrap;
+use Neos\Flow\Log\PsrLoggerFactoryInterface;
+use Neos\Setup\Domain\CliEnvironment;
+use Neos\Setup\Domain\EarlyBootTimeHealthcheckInterface;
+use Neos\Setup\Domain\Health;
+use Neos\Setup\Domain\HealthcheckEnvironment;
+use Neos\Setup\Domain\Status;
+use Neos\Utility\Files;
+use Neos\Setup\Infrastructure\HealthcheckFailedError;
+use Psr\Log\LoggerInterface;
+
+class BasicRequirementsHealthcheck implements EarlyBootTimeHealthcheckInterface
+{
+    private HealthcheckEnvironment $environment;
+
+    public function __construct(
+        private readonly ConfigurationManager $configurationManager,
+        private readonly LoggerInterface $logger
+    ) {
+    }
+
+    public static function fromBootstrap(Bootstrap $bootstrap): self
+    {
+        return new self(
+            $bootstrap->getEarlyInstance(ConfigurationManager::class),
+            $bootstrap->getEarlyInstance(PsrLoggerFactoryInterface::class)->get('systemLogger')
+        );
+    }
+
+    public function getTitle(): string
+    {
+        return 'Basic system requirements';
+    }
+
+    public function execute(HealthcheckEnvironment $environment): Health
+    {
+        $this->environment = $environment;
```

Comment:
i dont like it, but whats the alternative? @kitsunet 

and a `setEnviroment` method on the interface (or better like in other places with is_callable) is also not the holy grail ...

Reply:
Yeah, wouldn't do. Either we introduce options you can set per check and set those options differnetly per environment, or we just disable all checks in production (again via settings) and people decide what they want to activate.